test/components/generators/chat/test_azure.py

```diff
@@ -28,17 +42,19 @@ def test_init_fail_wo_api_key(self, monkeypatch):
         with pytest.raises(OpenAIError):
             AzureOpenAIChatGenerator(azure_endpoint="some-non-existing-endpoint")
 
-    def test_init_with_parameters(self):
+    def test_init_with_parameters(self, tools):
         component = AzureOpenAIChatGenerator(
             api_key=Secret.from_token("test-api-key"),
             azure_endpoint="some-non-existing-endpoint",
             streaming_callback=print_streaming_chunk,
             generation_kwargs={"max_tokens": 10, "some_test_param": "test-params"},
+            tools=tools,
```

Comment:
We could pass `tools_strict` here too, set it to `True` and then check it is set as expected.

Reply:
added